src/main/java/com/recurly/v3/Pager.java

```diff
@@ -93,4 +92,34 @@ private void clone(Pager<T> pager) {
     this.more = pager.hasMore();
     this.data = pager.getData();
   }
+
+  private class PagerIterator implements Iterator {
+    public PagerIterator(Object[] data, Pager<T> pager) {
+      this.data = data;
+      this.pager = pager;
+    }
+
+    private int position = 0;
+    private Object[] data;
+    private Pager<T> pager;
+
+    public boolean hasNext() {
+      return position < this.data.length || pager.hasMore();
+    }
+
+    public Object next() {
```

Comment:
I feel like this method could be cleaned up similarly, but I'm not really sure how. If you have any suggestions on making this more concise, I'm open.

Reply:
I think it's good now that we moved the hasNext check. Resolving this convo.